### PR TITLE
docs: add initial events deep dive

### LINKED ISSUE #1037: Add recommendation for event usage and firing
For usage of events have a TLDR.
- wire-up to local DOM via @some-event in template (let Lit do the actual wire-up)
- wire-up events to this in constructor (no need to clean up - as it will be garbage collected on destroying)
- wire-up to elements outside of scope (ex. window) using connectedCallback , and remove them in disconnectedCallback. 

For firing events have a TLDR.
- prefer "just" events without `bubble` or `composed` e.g. `new Event('something-happened')`
- when needing data consider extending `Event` or using `CustomEvent`
- use `bubble: true` if a node up the tree needs the event
- retrain from using `composed: true` as it leads to event pollution 
- use composed only if absolutely needed

Add a link for details too https://dev.to/open-wc/composed-true-considered-harmful-5g59

### PR DESCRIPTION
**Pull request by Westbrook**
Not wanting great to be the enemy of good, this is a first iteration for the event usage recommendations. Namely, I stole @daKmoR's great TLDR and dressed it up a bit. This should allow us to close #1037 and have something to start sharing as needed.

fixes #1037 
refs #895 

For #895, I'd like to get into the nuance of some of the recommendations and plan to expand on this with further exposition in a later iteration. If we're in agreement, let's get this out and then I can start a separate PR for us to discuss the additional content or code labs, etc.

### CODE REVIEW

File: docs/faq/README.md [hunk 1]
```diff
@@ -7,7 +7,6 @@ In this section you can find answers to frequently asked questions regarding jav
 
 **Deep dives:**
 
-- [Component Libraries](./component-libraries.html)
 - [Rerender not triggered](./rerender.html)
```
Comment (daKmoR) [inline docs/faq/README.md]: you wanna link to the events faq here right?

Comment (Westbrook) [reply docs/faq/README.md]: Interesting...how does something get into the `[[toc]]` annotation on these pages?

Comment (daKmoR) [reply docs/faq/README.md]: it's here
https://github.com/open-wc/open-wc/blob/master/docs/.vuepress/config.js#L128

Comment (Westbrook) [reply docs/faq/README.md]: Ok, this this iteration is good to go, now!

**Comment by daKmoR** [comment]
awesome thxxx 🤗

**Pull request merged**